visma/gui/cli.py

```diff
@@ -1,120 +1,218 @@
+            else:
+                lTokens, rTokens, _, _, equationTokens, comments = additionEquation(lTokens, rTokens, True)
+        elif operation == 'subtraction':
+            if solutionType == 'expression':
+                tokens, _, _, equationTokens, comments = subtraction(tokens, True)
+            else:
+                lTokens, rTokens, _, _, equationTokens, comments = subtractionEquation(lTokens, rTokens, True)
+        elif operation == 'multiplication':
+            if solutionType == 'expression':
+                tokens, _, _, equationTokens, comments = multiplication(tokens, True)
+            else:
+                lTokens, rTokens, _, _, equationTokens, comments = multiplicationEquation(lTokens, rTokens, True)
+        elif operation == 'division':
+            if solutionType == 'expression':
+                tokens, _, _, equationTokens, comments = division(tokens, True)
+            else:
+                lTokens, rTokens, _, _, equationTokens, comments = divisionEquation(lTokens, rTokens, True)
+        elif operation == 'factorize':
+            tokens, _, _, equationTokens, comments = factorize(tokens)
+        elif operation == 'find-roots':
+            lTokens, rTokens, _, _, equationTokens, comments = rootFinder(lTokens, rTokens)
+        elif operation == 'solve':
+            if simul:
+                if varName is not None:
+                    _, equationTokens, comments = simulSolver(tokens[0], tokens[1], tokens[2], varName)
+                else:
+                    _, equationTokens, comments = simulSolver(tokens[0], tokens[1], tokens[2])
+                solutionType = equationTokens
+            else:
+                lhs, rhs = getLHSandRHS(tokens)
+                lTokens, rTokens, _, _, equationTokens, comments = solveFor(lTokens, rTokens, varName)
+        elif operation == 'factorial':
+            tokens, _, _, equationTokens, comments = factorial(tokens)
+        elif operation == 'combination':
+            n = tokenizer(inputEquation)
+            r = tokenizer(varName)
+            tokens, _, _, equationTokens, comments = combination(n, r)
+        elif operation == 'permutation':
+            n = tokenizer(inputEquation)
+            r = tokenizer(varName)
+            tokens, _, _, equationTokens, comments = permutation(n, r)
+        elif operation == 'integrate':
+            lhs, rhs = getLHSandRHS(tokens)
+            lTokens, _, _, equationTokens, comments = integrate(lTokens, varName)
+        elif operation == 'differentiate':
             lhs, rhs = getLHSandRHS(tokens)
-            lTokens = lhs
-            rTokens = rhs
+            lTokens, _, _, equationTokens, comments = differentiate(lTokens, varName)
+        final_string = resultStringCLI(equationTokens, operation, comments, solutionType, simul)
+        print(final_string)
+    else:
+        operation = operation[4:]
+        dualOperand = False
```

Comment:
```suggestion
        # TODO: Add comments for below flags
        dualOperand = False
```

Make sure to add comments for these flags when you start commenting and documenting the code.